### PR TITLE
chmlib: Support Apple Silicon

### PR DESCRIPTION
**Pull request by nsmaciej**
The audit fails with the following, but I have no idea what to do about the license warning and there doesn't seem to be any text fixtures chm files for me to add new tests with.

```
chmlib:
  * 1: col 1: A `test do` test block should be added
  * Formula chmlib contains deprecated SPDX licenses: ["LGPL-2.1"].
    You may need to add `-only` or `-or-later` for GNU licenses (e.g. `GPL`, `LGPL`, `AGPL`, `GFDL`).
    For a list of valid licenses check: https://spdx.org/licenses/
```

---

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

File: Formula/chmlib.rb [hunk 1]
```diff
@@ -29,9 +29,26 @@ class Chmlib < Formula
...
++++ b/src/chm_lib.c
+@@ -164,7 +164,7 @@ typedef unsigned long long      UInt64;
+
+ /* x86-64 */
+ /* Note that these may be appropriate for other 64-bit machines. */
+-#elif __x86_64__ || __ia64__
++#elif __x86_64__ || __ia64__ || __aarch64__
+ typedef unsigned char           UChar;
+ typedef short                   Int16;
+ typedef unsigned short          UInt16;
```
Comment (carlocab) [inline Formula/chmlib.rb:54]: Has this patch been submitted upstream?

**Review by carlocab — COMMENTED** [review]
Thanks for the PR.

> there doesn't seem to be any text fixtures chm files for me to add new tests with.

Can we try to do something with a non-existent chm file and check we get the error message and exit code we should be getting?

**Comment by carlocab** [comment]
This is `LGPL-2.1-or-later`, as indicated in the headers of the source files. For example, in `chm_lib.c` and `chm_lib.h`:
```
/***************************************************************************
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU Lesser General Public License as        *
 *   published by the Free Software Foundation; either version 2.1 of the  *
 *   License, or (at your option) any later version.                       *
 *                                                                         *
 ***************************************************************************/
```

File: Formula/chmlib.rb (hunk 1 above)
Comment (mgoszcz2) [reply Formula/chmlib.rb:66]: I haven't tried, but considering there haven't been any commits in 13 years it seemed a bit moot.

**Comment by mgoszcz2** [comment]
Unfortunately chmlib is not very informative when it comes to errors, `chm_open` is either successful or it returns NULL.
I added a simple test for that.

**Comment by mgoszcz2** [comment]
Out of curiosity, what does "CI-no-bottles" imply?

**Review by carlocab — APPROVED** [review]
Thanks!

**Comment by carlocab** [comment]
> Out of curiosity, what does "CI-no-bottles" imply?

It tells @BrewTestBot not to publish bottles if they merge this PR automatically. We don't want that as that discards all the older bottles. I've dispatched a workflow to add the ARM bottles separately to preserve the older ones:

https://github.com/Homebrew/homebrew-core/actions/runs/1685003961

**Pull request merged**